RecoTracker/TrackProducer/python/VertexConstraintProducer_cff.py

```diff
@@ -0,0 +1,7 @@
+import FWCore.ParameterSet.Config as cms
+
+VertexConstraint = cms.EDProducer(
```

Comment:
This module defines the default configuration for the VertexConstraintProducer: it should therefore be a ```_cfi```, not a ```_cff```

Reply:
Following the CMS rules, the module ```VertexConstraint``` should start with a lower-case letter: ```vertexConstraint```?

Reply:
But mainly: the class VertexConstraintProducer is coded into a \test area:
```
RecoTracker/TrackProducer/test/VertexConstraintProducer.cc
```
As it is going to be used in Alignment, shouldn't it get moved into a \plugin area (perhaps, together with the GsfVertexConstraintProducer)?

Reply:
- ```VertexConstraintProducer_cff.py``` -> ```VertexConstraintProducer_cfi.py```: this is a new file made by you. Please rename, and follow the same convention as other configs in the same directory, e.g ```TrackRefitter_cfi.py``` (where the _cff fragments also exist and correctly derive from it). The ```TwoBodyDecayConstraintProducer_cff.py```  that you point out was indeed wrongly named: it should eventually get fixed, but I don't ask you to do so in this PR.

- ```VertexConstraint``` -> ```vertexConstraint``` : I see that there are a few examples in this \python area where the module names start with a uppercase letter (but there are also examples of the contrary). Again, since this is some new code which is going to be integrated  for the first time, I don't see why we shouldn't follow the CMS convention

- Plugins in the \test area. This indeed is a legacy of the code already in CMSSW. I think that it should better get fixed at some point, but maybe we can postpone it to another pull request, if deemed necessary

Reply:
@hroskes : please let us know when you plan to apply the small fixes requested, so that we can proceed with this PR